File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/request/FileInput.java
package com.ashlikun.okhttputils.http.request;

import com.ashlikun.okhttputils.http.HttpUtils;

import java.io.File;

import okhttp3.MediaType;

/**
 * 作者　　: 李坤
 * 创建时间:2017/3/17　15:56
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：
 */

public class FileInput {
    public String key;
    public String filename;
    public MediaType contentType;
    public File file;

    public FileInput(String key, String filename, File file) {
        this.key = key;
        this.filename = filename;
        this.file = file;
        if (exists()) {
            contentType = MediaType.parse(HttpUtils.getMimeType(file.getAbsolutePath()));
        }
    }

    public FileInput(String key, File file) {
        this.key = key;
        this.file = file;
        if (exists()) {
            this.filename = file.getName();
            contentType = MediaType.parse(HttpUtils.getMimeType(file.getAbsolutePath()));
        }

    }

    public boolean exists() {
        try {
            return file.exists();
        } catch (Exception e) {
            e.printStackTrace();
        }

        return false;
    }

    @Override
    public String toString() {
        return "FileInput{" +
                "key='" + key + '\'' +
                ", filename='" + filename + '\'' +
                ", contentType='" + contentType + '\'' +
                ", file=" + file +
                '}';
    }

}

File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/request/FileInput.kt
package com.ashlikun.okhttputils.http.request

import com.ashlikun.okhttputils.http.HttpUtils.getMimeType
import okhttp3.MediaType
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.MediaType.Companion.toMediaTypeOrNull
import java.io.File
import java.lang.Exception

/**
 * 作者　　: 李坤
 * 创建时间:2017/3/17　15:56
 * 邮箱　　：496546144@qq.com
 *
 *
 * 功能介绍：
 */
class FileInput(
    var key: String,
    var file: File,
    var filename: String = "",
    var contentType: MediaType? = null
) {

    init {
        if (exists()) {
            if (filename.isEmpty()) {
                filename = file.name
            }
            if (contentType == null) {
                contentType = getMimeType(file.absolutePath).toMediaTypeOrNull()
            }
        }
    }

    fun exists(): Boolean {
        try {
            return file.exists()
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return false
    }

    override fun toString(): String {
        return "FileInput{" +
                "key='" + key + '\'' +
                ", filename='" + filename + '\'' +
                ", contentType='" + contentType + '\'' +
                ", file=" + file +
                '}'
    }
}